util/kube/kube.go

```diff
@@ -107,6 +107,56 @@ func MustToUnstructured(obj interface{}) *unstructured.Unstructured {
 	return uObj
 }
 
+// UnsetLabel removes our app labels from an unstructured object
+func UnsetLabel(target *unstructured.Unstructured, key string) error {
+	labels := target.GetLabels()
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+	delete(labels, key)
+	target.SetLabels(labels)
+
+	// special case for deployment types: make sure that derived replicaset and pod has application label
+	switch target.GetKind() {
```

Comment:
This switch block appears to be just copied and pasted over from the SetLabels logic -- the comments don't even make sense here. We actually don't need to unset the nested pod template labels since ksonnet sets the labels at the object.metadata level instead of the pod template level. Lets remove the switch entirely.

After removing the switch block, please also remove the return value since it will be impossible to return error.